Feature: new wallet flow

adrian-tiberius opened the pull request:
Implements new wallet flow.

rasom commented:
1.  `request-transaction` label

![screen shot 2016-11-01 at 12 00 23](https://cloud.githubusercontent.com/assets/2364994/19887549/c6f80aae-a031-11e6-8d00-a4ea48a7c386.png)

2. Maybe it is better to use `?` when it is an answer to the request and `!` when it is just a command

![screen shot 2016-11-01 at 12 02 15](https://cloud.githubusercontent.com/assets/2364994/19887578/f1aca232-a031-11e6-96b5-5d9c07e920bb.png)

3. 

![screen shot 2016-11-01 at 12 04 49](https://cloud.githubusercontent.com/assets/2364994/19887634/3547ec68-a032-11e6-85aa-1830d7fe916d.png)

4. Can not scan QR

![qr](https://cloud.githubusercontent.com/assets/2364994/19887674/853a1908-a032-11e6-9c21-2e78e831a558.gif)

rasom commented:
These two bugs are not related to this PR, but they appear because of it :) 
I will create separate issues

1. 

![screen shot 2016-11-01 at 12 03 41](https://cloud.githubusercontent.com/assets/2364994/19887617/23138a2a-a032-11e6-8c64-0b95dc3ce330.png)


2. Request icon is overlapped by the next message 
![screen shot 2016-11-01 at 12 38 38](https://cloud.githubusercontent.com/assets/2364994/19887658/669d8548-a032-11e6-8129-8f0e06577bd9.png)

jarradh commented:
regarding 1, good question. I don't like the idea of truncating the name but we will have to. 

and 2 (1?) I guess request message should have a minimum height that eliminates the overlap

jarradh commented:
both the request message and the send eth message will have better designs

jarradh commented:
for example, at the bottom of this https://zpl.io/Z1cQdF4

jarradh commented:
regarding earlier comment:
- yes !send should be !send.. not sure why this has changed, it's not using the !send command ?
- yes the parameters should be prefilled
- although we can think of a better name than `request-transaction` we should establish a limit., I also notice that we're missing the 'pretty' command titles and instead still filling with the !command name

adrian-tiberius commented:
- changed request-transaction to request (this doesn't solve the long command names problem which I think needs to be discussed in a different issue about dapp api specs)
- response prefill is related to general handling of requests/responses and I think it would be cleaner to tackle it in a different issue/pr
- fixed qr scan

alwx commented on src/status_im/accounts/styles.cljs:
```diff
@@ -113,3 +113,56 @@
+   :right    0
+   :bottom   0
+   :height 166
+   :background-color "#2f3031"})
```
code formatting for hashmaps is a bit strange, but it is not a big deal, of course

alwx commented on src/status_im/chat/handlers/requests.cljs:
```diff
@@ -15,6 +16,7 @@
                  :type       (:command content)
                  :added      (js/Date.)}
         request' (update request :type keyword)]
+    (log/debug "Adding request: " request')
```
do we need all these log/debug calls everywhere?

adrian-tiberius replied:
will be removed before squashing if everything is ok.

alwx commented on src/status_im/data_store/test_data/contacts.cljs:
```diff
@@ -0,0 +1,42 @@
+  (when-not (data-store/exists? "test_identity_1")
+    (doseq [contact contacts]
+      (log/debug "Adding contact " contact)
+      (data-store/save contact false))))
```
Do we need the test data to be merged to develop branch?

jarradh replied:
no

rasom commented:
I see this error after scanning QR

![screenshot_2016-11-04-17-14-10](https://cloud.githubusercontent.com/assets/2364994/20010839/45b26d58-a2b2-11e6-9a13-6ddf119d171d.png)

rasom commented:
Send request

<img width="426" alt="screen shot 2016-11-04 at 16 53 58" src="https://cloud.githubusercontent.com/assets/2364994/20010969/bd041834-a2b2-11e6-8212-90fd590020e2.png">

but receive response

![screenshot_2016-11-04-16-53-55](https://cloud.githubusercontent.com/assets/2364994/20010990/cbd33282-a2b2-11e6-9195-6e197281d41f.png)
